placement/staged_placement_watcher.go

```diff
@@ -139,14 +144,23 @@ func (t *stagedPlacementWatcher) toStagedPlacement(value kv.Value) (interface{},
 	return NewStagedPlacementFromProto(version, t.proto, t.placementOpts)
 }
 
-func (t *stagedPlacementWatcher) process(value interface{}) error {
+func (t *stagedPlacementWatcher) process(update interface{}) error {
 	t.Lock()
 	defer t.Unlock()
 
 	if t.state != placementWatcherWatching {
 		return errPlacementWatcherIsNotWatching
 	}
-	ps := value.(StagedPlacement)
+	if update == nil {
```

Comment:
I think by the time you get here, `update` is guaranteed to be non nil, yeah?

Reply:
That is depending on the `GetFn` passed in by user, if the user does bad things then here can still be nil, so I'd rather be on the defensive side.

Reply:
Hm why would it depend on `GetFn`? It seems `processFn` is only called from `updateWithLock` in `value.go` and the first line in that function is checking whether the update is nil?

Reply:
Because the if the user pass in a GetFn that returns `nil, nil` then the update would be nil

Reply:
Sure but that should be caught by the nil check on the first line of `updateWithLock` I'd think?

Reply:
yeah that is true, can remove this